src/go/rpk/pkg/cli/redpanda/mode.go

```diff
@@ -63,3 +63,40 @@ func executeMode(fs afero.Fs, p *config.Params, mode string) error {
 	}
 	return cfg.SetMode(fs, mode)
 }
+
+const modeHelpText = `Enable a default configuration mode
+
+This command allows you to set one of the following modes: Development,
+Production, or Recovery.
+
+PRODUCTION
+
+Disables dev-mode settings:
+
+  - developer_mode: false
+  - overprovisioned: false
+
+It also enables a set of tuners for the rpk tuner. For descriptions about the 
+tuners, check the 'rpk redpanda tune' command.
+
+DEVELOPMENT
+
+Sets 'redpanda.developer_mode=true' in your redpanda.yaml file. This starts 
+Redpanda with dev-mode only settings, including:
```

Comment:
Sorry, I missed this the first time. redpanda.developer_mode doesn't do much or really anything, and it's not worth going into detail here. Maybe something like "Allows redpanda to bypass some internal optimal runtime checks". We also want to mention that development disables tuners (so, keep the final bullet point).

The mode=dev-container is what bypasses fsync and disables overprovisioning, and this is done via rpk itself.

Reply:
Err well, looks like we do set overprovisioned=false (looking at what you mentioned in the production bit above). I'm still not sure it's worth going into detail here though, since it'll be hard to ensure we update documentation if we ever change what development vs. production mode does. I'd only have two bullet points, one mentioning bypassing or disabling optimal runtime checks, and the other mentioning tuners.